### ISSUE TITLE
`let closure = { Void -> Void in }` produces "Variable Name Violation"

### ISSUE THREAD

**Issue by norio-nomura** [realm/SwiftLint#570]
e.g:

``` sh
% echo "let closure = { Void -> Void in }"|swiftlint lint --use-stdin
<nopath>:1:17: error: Variable Name Violation: Variable name should start with a lowercase character: 'Void' (variable_name)
Done linting! Found 1 violation, 1 serious in 1 file.
()
```

This is caused by that SourceKitService returns  `"key.name" : "Void"` as following: 

``` sh
% sourcekitten structure --text "let closure = { Void -> Void in }"
{
  "key.substructure" : [
    {
      "key.kind" : "source.lang.swift.decl.var.global",
      "key.offset" : 0,
      "key.nameoffset" : 4,
      "key.namelength" : 7,
      "key.length" : 33,
      "key.accessibility" : "source.lang.swift.accessibility.internal",
      "key.typename" : "closure",
      "key.name" : "closure"
    },
    {
      "key.kind" : "source.lang.swift.decl.var.parameter",
      "key.offset" : 16,
      "key.nameoffset" : 0,
      "key.namelength" : 0,
      "key.length" : 4,
      "key.typename" : "Void",
      "key.name" : "Void"
    },
    {
      "key.kind" : "source.lang.swift.stmt.brace",
      "key.offset" : 14,
      "key.nameoffset" : 0,
      "key.namelength" : 0,
      "key.bodyoffset" : 15,
      "key.bodylength" : 17,
      "key.length" : 19
    }
  ],
  "key.offset" : 0,
  "key.diagnostic_stage" : "source.diagnostic.stage.swift.parse",
  "key.length" : 33
}
```

**Comment by jpsim** [comment]
This one might be tricky for SourceKit to fix, because the syntax would actually allow either a type or a variable name in that context.

**Comment by norio-nomura** [comment]
Yeah, we talked about this issue on slack and that might be acceptable special case to assume that `Void` is not really the `decl.var.parameter`.

**Comment by jpsim** [comment]
We're kind of lucky in that we now have another rule that discourages the use of `Void ->` which would also report a violation here, so even though this is technically a bug, I think it's fine to leave as-is.
